File: 01-50/10/10.c
#include <stdint.h>
#include <stdio.h>
#include <stdlib.h>

uint64_t sieve(int limit) {
    int *sieve = calloc(1, limit * sizeof(int));
    uint64_t total = 0;
    for (int i = 2; i < limit; i += 1) {
        if (sieve[i] == 1) {
            continue;
        } else {
            total += i;
            for (int j = i; j < limit; j += i) {
                sieve[j] = 1;
            }
        }
    }
    return total;
}

int main() {
    uint64_t total = sieve(2000000);
    printf("total: %llu\n", total);
}

File: 01-50/10/10.py
def sieve(limit):
    bits = [0] * (limit + 1)
    for i in xrange(2, limit):
        if bits[i] == -1:
            continue
        yield i
        bits[i] = 1
        for j in xrange(i, limit, i):
            bits[j] = -1

print sum(sieve(2000000))
